python/ray/data/dataset.py

```diff
@@ -2380,6 +2380,27 @@ def to_tf(
         if isinstance(output_signature, list):
             output_signature = tuple(output_signature)
 
+        def get_df_values(df: "pandas.DataFrame") -> np.ndarray:
+            # TODO(Clark): Support unsqueezing column dimension API, simialr to
+            # to_torch().
+            try:
+                values = df.values
+            except ValueError as e:
+                import pandas as pd
+
+                # Pandas DataFrame.values doesn't support extension arrays in all
+                # supported Pandas versions, so we check to see if this DataFrame
+                # contains any extensions arrays and do a manual conversion if so.
+                # See https://github.com/pandas-dev/pandas/pull/43160.
+                if any(
+                    isinstance(dtype, pd.api.extensions.ExtensionDtype)
+                    for dtype in df.dtypes
+                ):
+                    values = np.stack([col.to_numpy() for _, col in df.items()], axis=1)
```

Comment:
hmm do we want to conver all columns to numpy if _any_ column is an ExtensionDtype.

Or should we only do col.to_numpy() only if the dtype of that particular columns is an ExtensionDtype

Reply:
It doesn't matter, `df.values` converts all columns to NumPy ndarrays anyway so it would be exactly the same.